### DISCUSSION TITLE
Upload multiple programs

### DISCUSSION THREAD

**Discussion by SAqibDjinn** [visose/Robots#127] [Q&A] [answered]
Hey, 
I am developing a script where I need to change the properties of the tools TCP a few times for fabrication. Changing the TCP during the uploaded program causes the issue (Using UR10 - Path sanity check failed).  So instead I thought I could create multiple programs without changing the TCP (Using the create Program Component) and am now looking at how to upload them all at once in order for the UR 10 to play let's say 8 programs after each other in one go. Is something like this possible (Maybe with the custom code component)? 
Any help would be appreciated.
Best Saqib

**Comment by visose** [comment, marked as answer]
I would first check why are you getting the error. Usually it's due to a large movement or a change that the robot cannot handle.
When you mention you are changing the TCP of the tool, what do you mean exactly? For example are different targets of the same program using different TCPs, or you are running the same program multiple times with a different TCP? If its the latter, make sure you have a joint target at the start of each sequence so that the robot can reset to the same place no mater its pose at the end of the sequence.

I doubt this is the issue, but since you are using a script, make sure to use the latest version 1.0.4, as I recently made the TCP property setter private. If you want to assign different TCPs to different targets, make sure you use a unique tool for each TCP, don't change the TCP property as changing it will also affect any previous targets that have the same tool.

In general, don't mutate any target attributes, make new instances.
